latch/types/directory.py

```diff
@@ -93,6 +110,8 @@ def __init__(
         else:
             self.path = str(path)
 
+        self._path_witness = False
```

Comment:
any difference between this and `self.path is not None`?

Reply:
```suggestion
        self._path_generated = False
```

Reply:
`self.path` is set in `__init__` (this needs to change at some point)